src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -573,85 +573,64 @@ class spell_pri_mind_bomb : public AuraScript
+        return ValidateSpellInfo
+        ({
+            SPELL_PRIEST_PENANCE_DAMAGE,
+            SPELL_PRIEST_PENANCE_HEALING
+        });
+    }
 
-                uint8 rank = spellInfo->GetRank();
-                if (!sSpellMgr->GetSpellWithRank(SPELL_PRIEST_PENANCE_R1_DAMAGE, rank, true))
-                    return false;
-                if (!sSpellMgr->GetSpellWithRank(SPELL_PRIEST_PENANCE_R1_HEAL, rank, true))
-                    return false;
+    bool Load() override
+    {
+        return GetCaster()->GetTypeId() == TYPEID_PLAYER;
+    }
 
-                return true;
-            }
+    SpellCastResult CheckCast()
+    {
+        Player* caster = GetCaster()->ToPlayer();
 
-            void HandleDummy(SpellEffIndex /*effIndex*/)
+        if (Unit* target = GetExplTargetUnit())
+        {
+            if (!caster->IsFriendlyTo(target))
             {
-                Unit* caster = GetCaster();
-                if (Unit* target = GetHitUnit())
-                {
-                    if (!target->IsAlive())
-                        return;
-
-                    uint8 rank = GetSpellInfo()->GetRank();
+                if (!caster->IsValidAttackTarget(target))
+                    return SPELL_FAILED_BAD_TARGETS;
 
-                    if (caster->IsFriendlyTo(target))
-                        caster->CastSpell(target, sSpellMgr->GetSpellWithRank(SPELL_PRIEST_PENANCE_R1_HEAL, rank), false);
-                    else
-                        caster->CastSpell(target, sSpellMgr->GetSpellWithRank(SPELL_PRIEST_PENANCE_R1_DAMAGE, rank), false);
-                }
+                if (!caster->isInFront(target))
+                    return SPELL_FAILED_UNIT_NOT_INFRONT;
             }
+        }
 
-            SpellCastResult CheckCast()
-            {
-                Player* caster = GetCaster()->ToPlayer();
-                if (Unit* target = GetExplTargetUnit())
-                {
-                    if (!caster->IsFriendlyTo(target))
-                    {
-                        if (!caster->IsValidAttackTarget(target))
-                            return SPELL_FAILED_BAD_TARGETS;
-
-                        if (!caster->isInFront(target))
-                            return SPELL_FAILED_UNIT_NOT_INFRONT;
-                    }
-                }
-
-                return SPELL_CAST_OK;
-            }
+        return SPELL_CAST_OK;
+    }
 
-            void Register() override
-            {
-                OnEffectHitTarget += SpellEffectFn(spell_pri_penance_SpellScript::HandleDummy, EFFECT_0, SPELL_EFFECT_DUMMY);
-                OnCheckCast += SpellCheckCastFn(spell_pri_penance_SpellScript::CheckCast);
-            }
-        };
+    void HandleDummy(SpellEffIndex /*effIndex*/)
+    {
+        Unit* caster = GetCaster();
 
-        SpellScript* GetSpellScript() const override
+        if (Unit* target = GetHitUnit())
         {
-            return new spell_pri_penance_SpellScript;
+            if (!target->IsAlive())
```

Comment:
`SPELL_ATTR2_CAN_TARGET_DEAD` isn't set for 47540, so we shouldn't reach this point i think?

Reply:
nevermind, this cast is done while channeling (we are literally causing the waves here)

Reply:
this is the dummy, the triggered spells are the channeled spells, the individual bolts are different Ids